internal/provider/app_owner_resource.go

```diff
@@ -0,0 +1,241 @@
+}
+
+func (r *AppOwnerResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var data *AppOwnerResourceModel
+	var item types.Object
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &item)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	resp.Diagnostics.Append(item.As(ctx, &data, basetypes.ObjectAsOptions{
+		UnhandledNullAsEmpty:    true,
+		UnhandledUnknownAsEmpty: true,
+	})...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	var owners []shared.User
+	pageToken := ""
+	for {
+		listResp, err := r.client.AppOwners.C1APIAppV1AppOwnersList(ctx, operations.C1APIAppV1AppOwnersListRequest{
+			AppID:     data.AppID.ValueString(),
+			PageToken: &pageToken,
+		})
+		if err != nil {
+			resp.Diagnostics.AddError("failure to invoke list API", err.Error())
+			return
+		}
+		if listResp == nil {
+			resp.Diagnostics.AddError("unexpected response from list API", fmt.Sprintf("%v", listResp))
+			return
+		}
+		if listResp.ListAppOwnersResponse == nil || listResp.ListAppOwnersResponse.List == nil {
+			resp.Diagnostics.AddError("unexpected response from list API", fmt.Sprintf("%v", listResp))
+			return
+		}
+
+		owners = append(owners, listResp.ListAppOwnersResponse.List...)
+
+		if listResp.ListAppOwnersResponse.NextPageToken == nil || *listResp.ListAppOwnersResponse.NextPageToken == "" {
+			break
+		}
+		pageToken = *listResp.ListAppOwnersResponse.NextPageToken
+	}
+
+	data.RefreshFromReadResponse(owners)
+
+	// Save updated data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *AppOwnerResource) Update(ctx context.Context, req resource.UpdateRequest, resp *resource.UpdateResponse) {
+	var data *AppOwnerResourceModel
+	merge(ctx, req, resp, &data)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	// Not Implemented; all attributes marked as RequiresReplace
+
+	// Save updated data into Terraform state
+	resp.Diagnostics.Append(resp.State.Set(ctx, &data)...)
+}
+
+func (r *AppOwnerResource) Delete(ctx context.Context, req resource.DeleteRequest, resp *resource.DeleteResponse) {
+	var data *AppOwnerResourceModel
+	var item types.Object
+
+	resp.Diagnostics.Append(req.State.Get(ctx, &item)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	resp.Diagnostics.Append(item.As(ctx, &data, basetypes.ObjectAsOptions{
+		UnhandledNullAsEmpty:    true,
+		UnhandledUnknownAsEmpty: true,
+	})...)
+
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	setAppOwnersRequest := data.ToDeleteSDKType()
+	appID := data.AppID.ValueString()
+	request := operations.C1APIAppV1AppOwnersSetRequest{
+		SetAppOwnersRequest: setAppOwnersRequest,
+		AppID:               appID,
+	}
+	res, err := r.client.AppOwners.C1APIAppV1AppOwnersSet(ctx, request)
+	if err != nil {
+		resp.Diagnostics.AddError("failure to invoke API", err.Error())
+		return
+	}
+	if res == nil {
+		resp.Diagnostics.AddError("unexpected response from API", fmt.Sprintf("%v", res))
```

Comment:
C1APIAppV1AppOwnersSet() already throws an error when res is nil, this code is unreachable